src/ice_shelf/MOM_ice_shelf.F90

```diff
@@ -772,39 +805,54 @@ end subroutine change_thickness_using_melt
 
 !> This subroutine adds the mechanical forcing fields and perhaps shelf areas, based on
 !! the ice state in ice_shelf_CS.
-subroutine add_shelf_forces(G, US, CS, forces, do_shelf_area)
```

Comment:
The forces and fluxes types use the ocean's grid for their halo sizes and indexing conventions, which may at some point be different from what is used in the ice-shelf code. The G argument with the ocean's grid structure should not have been removed here or in the subsequent interface to `add_shelf_pressure()`.  Adding a grid to the ice_shelf_CS is fine, but if there is a separate ice shelf grid, `add_shelf_forces()` and `add_shelf_pressure()` need to be written in a way that takes into account the fact that these two grids may be different!

Reply:
The fluxes and forcing types use the unrotated grid  which is pointed to by the shelf structure, so I do not believe we need to retain it in the interface.   The ice shelf is creating a grid using the same parameter file as MOM6. If we were to use a separate grid and PF for the shelf, then we could use the ocean grid pointer which is still an argument to initialize_ice_shelf.

Reply:
I returned the ocean grid to the interface to add_shelf_forces and included a grid compatibility check